malformed/test_malformed_crlfs.py

```diff
@@ -0,0 +1,438 @@
+    	#
+        # This is a heavy chunked version of the test
+        #
+        request = \
+                  'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n' \
+                  '\n' \
+                  'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 200, expect, True)
+
+    def test_17_double_crlf_hch(self):
+    	# Test double CRLF before request
+    	# Request should be rejected by the proxy
+    	#
+        # This is a heavy chunked version of the test
+        #
+        request = \
+                  '\r\n' \
+                  '\r\n' \
+                  'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = None
+        self.common_check(request, 400, expect, True)
+
+    def test_18_double_crlf_pipeline_hch(self):
+    	# Test double CRLF before 2nd request in a pipeline
+    	# The 1st request should be passed to backend
+    	# The 2nd request should be rejected by the proxy
+    	#
+        # This is a heavy chunked version of the test
+        #
+        request = \
+                  'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n' \
+                  '\r\n' \
+                  '\r\n' \
+                  'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 400, expect, True)
+
+    def test_19_double_lf_hch(self):
+    	# Test double LF before request
+    	# Request should be rejected by the proxy
+    	#
+        # This is a heavy chunked version of the test
+        #
+        request = \
+                  '\n' \
+                  '\n' \
+                  'GET / HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = None
+        self.common_check(request, 400, expect, True)
+
+    def test_20_double_lf_pipeline_hch(self):
+    	# Test double LF before 2nd request in a pipeline
+    	# The 1st request should be sent to backed
+    	# The 2nd request should be rejected by the proxy
+    	#
+        # This is a heavy chunked version of the test
+        #
+        request = \
+                  'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n' \
+                  '\n' \
+                  '\n' \
+                  'GET /bbb HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        expect = 'GET /aaa HTTP/1.1\r\n' \
+                  'Host: localhost\r\n' \
+                  '\r\n'
+        self.common_check(request, 400, expect, True)
```

Comment:
I like the tests a lot: the tests itself, the example of using chunks and the documentation in the code!